### PR TITLE
feature:关联 namespace界面，可以一次选择多个namespace

### PR DESCRIPTION
**Pull request by falser101**
Signed-off-by: jianfei.zhang jianfei.zhang@daocloud.io

## What's the purpose of this PR

关联 namespace界面，可以一次选择多个namespace

## Which issue(s) this PR fixes:
Fixes # #4329 

Follow this checklist to help us incorporate your contribution quickly and easily:
- [x] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run mvn clean test to make sure this pull request doesn't break anything.
- [x] Update the [CHANGES log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by falser101** [comment]
recheck

**Comment by falser101** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by falser101** [comment]
recheck

**Comment by falser101** [comment]
image

File: apollo-portal/src/main/resources/static/scripts/controller/NamespaceController.js [hunk 1]
```diff
@@ -94,32 +94,38 @@ namespace_module.controller("LinkNamespaceController",
...
                         return;
                     }
 
-                    if ($scope.namespaceType == 1) {
-                        var selectedNamespaceName = $('#namespaces').select2('data')[0].id;
-                        if (!selectedNamespaceName) {
+                    if ($scope.namespaceType === 1) {
+                        var selectedNamespaceNames = $('#namespaces').select2('data');
+                        var ids = []
+                        selectedNamespaceNames.map().forEach(function (namespace) {
```
Comment (klboke) [inline apollo-portal/src/main/resources/static/scripts/controller/NamespaceController.js:106]: `selectedNamespaceNames.map()` will throw the following exception：
```js
TypeError: undefined is not a function
    at Array.map (<anonymous>)
    at $scope.createNamespace (NamespaceController.js:110:48)
    at fn (eval at compile (angular.min.js:224:319), <anonymous>:4:236)
    at b (angular.min.js:121:504)
    at e (angular.min.js:266:74)
    at b.$eval (angular.min.js:140:467)
    at b.$apply (angular.min.js:141:188)
    at HTMLFormElement.<anonymous> (angular.min.js:266:126)
    at Lf (angular.min.js:36:394)
    at HTMLFormElement.d (angular.min.js:36:341)
```
An array can be iterated as follows:
```js
angular.forEach(selectedNamespaceNames, function(item){
                            ids.push(item.id)
                        });
```

Comment (falser101) [reply apollo-portal/src/main/resources/static/scripts/controller/NamespaceController.js]: i am so sorry, my local env have some trouble

**Comment by falser101** [comment]
image
image

@klboke @nobodyiam

**Review by nobodyiam — CHANGES_REQUESTED** [review]
There is a Permission Management page after the namespaces are associated. This was designed to let the user assign the permissions for the associated namespace.
However, since now we allow users to select multiple namespaces, it's not easy to let the user assign the permissions for multiple namespaces. But currently, it will display undefined as shown in the following screenshot, which is a bug. We need to adapt for this kind of scenario, e.g. if multiple namespaces are selected, simply redirect to the project page.

image

**Comment by falser101** [comment]
That's a good point. Let me revise it

**Comment by falser101** [comment]
> ‎命名空间关联后，将出现“权限管理”页。这旨在让用户为关联的命名空间分配权限。‎‎但是，由于现在我们允许用户选择多个命名空间，因此让用户为多个命名空间分配权限并不容易。但目前，它将显示未定义，如以下屏幕截图所示，这是一个错误。我们需要适应这种情况，例如，如果选择了多个命名空间，只需重定向到项目页面即可。‎
> 
> image

image


> There is a Permission Management page after the namespaces are associated. This was designed to let the user assign the permissions for the associated namespace. However, since now we allow users to select multiple namespaces, it's not easy to let the user assign the permissions for multiple namespaces. But currently, it will display undefined as shown in the following screenshot, which is a bug. We need to adapt for this kind of scenario, e.g. if multiple namespaces are selected, simply redirect to the project page.
> 
> image

now, redirect to the thispage
image

File: CHANGES.md [hunk 2]
```diff
@@ -13,5 +13,6 @@ Apollo 2.1.0
 * [Upgrade mysql-connector-java version to fix possible transaction rollback failure issue](https://github.com/apolloconfig/apollo/pull/4425)
 * [Remove database migration tool Flyway](https://github.com/apolloconfig/apollo/pull/4361)
 * [Optimize Spring-Security Firewall Deny Request Response 400](https://github.com/apolloconfig/apollo/pull/4428)
+* [On the Namespace association screen, you can select multiple namespaces at a time](https://github.com/apolloconfig/apollo/pull/4437)
```
Comment (nobodyiam) [inline CHANGES.md:16]:
```suggestion
* [Allow users to associate multiple public namespaces at a time](https://github.com/apolloconfig/apollo/pull/4437)
```

**Comment by nobodyiam** [comment]
> now, if user select multiple namespaces, it will redirect to the this page image else if user select one namespace,it will redirect to Permission Management page image @nobodyiam

This looks fine to me.

**Review by nobodyiam — APPROVED** [review]
LGTM
@klboke please also help to take a look, thx.

**Review by klboke — APPROVED** [review]
LGTM

**Pull request merged**